From 77380dade5f8214056a98b3d7586483ad9268ac6 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Wed, 2 Oct 2024 16:00:33 +0200
Subject: [PATCH] arm64: dts: st: add HSI/HSE/MSI Kernel parent clocks on
 stm32mp23

HSI/HSE/MSI Kernel parent clocks are SCMI clocks and need for RCC driver.

Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: Ic634018e9501a7d1bf7f5407053bb71929679e22
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/408551
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp231.dtsi |  6 +-
 arch/arm64/boot/dts/st/stm32mp235.dtsi | 87 ++++++++++++++++++++++++++
 2 files changed, 92 insertions(+), 1 deletion(-)

--- a/arch/arm64/boot/dts/st/stm32mp231.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp231.dtsi
@@ -2265,7 +2265,11 @@
 				<&scmi_clk CK_SCMI_ICN_APBDBG>,
 				<&scmi_clk CK_SCMI_TIMG1>,
 				<&scmi_clk CK_SCMI_TIMG2>,
-				<&scmi_clk CK_SCMI_PLL3>;
+				<&scmi_clk CK_SCMI_PLL3>,
+				<0>,
+				<&scmi_clk CK_SCMI_HSI_KER_CK>,
+				<&scmi_clk CK_SCMI_HSE_KER_CK>,
+				<&scmi_clk CK_SCMI_MSI_KER_CK>;
 			access-controllers = <&rifsc 156>;
 		};
 
--- a/arch/arm64/boot/dts/st/stm32mp235.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp235.dtsi
@@ -32,6 +32,93 @@
 	clock-names = "bus", "ref", "lcd", "lvds";
 };
 
+&rcc {
+	clocks =
+		<&scmi_clk CK_SCMI_HSE>,
+		<&scmi_clk CK_SCMI_HSI>,
+		<&scmi_clk CK_SCMI_MSI>,
+		<&scmi_clk CK_SCMI_LSE>,
+		<&scmi_clk CK_SCMI_LSI>,
+		<&scmi_clk CK_SCMI_HSE_DIV2>,
+		<&scmi_clk CK_SCMI_ICN_HS_MCU>,
+		<&scmi_clk CK_SCMI_ICN_LS_MCU>,
+		<&scmi_clk CK_SCMI_ICN_SDMMC>,
+		<&scmi_clk CK_SCMI_ICN_DDR>,
+		<&scmi_clk CK_SCMI_ICN_DISPLAY>,
+		<&scmi_clk CK_SCMI_ICN_HSL>,
+		<&scmi_clk CK_SCMI_ICN_NIC>,
+		<&scmi_clk CK_SCMI_ICN_VID>,
+		<&scmi_clk CK_SCMI_FLEXGEN_07>,
+		<&scmi_clk CK_SCMI_FLEXGEN_08>,
+		<&scmi_clk CK_SCMI_FLEXGEN_09>,
+		<&scmi_clk CK_SCMI_FLEXGEN_10>,
+		<&scmi_clk CK_SCMI_FLEXGEN_11>,
+		<&scmi_clk CK_SCMI_FLEXGEN_12>,
+		<&scmi_clk CK_SCMI_FLEXGEN_13>,
+		<&scmi_clk CK_SCMI_FLEXGEN_14>,
+		<&scmi_clk CK_SCMI_FLEXGEN_15>,
+		<&scmi_clk CK_SCMI_FLEXGEN_16>,
+		<&scmi_clk CK_SCMI_FLEXGEN_17>,
+		<&scmi_clk CK_SCMI_FLEXGEN_18>,
+		<&scmi_clk CK_SCMI_FLEXGEN_19>,
+		<&scmi_clk CK_SCMI_FLEXGEN_20>,
+		<&scmi_clk CK_SCMI_FLEXGEN_21>,
+		<&scmi_clk CK_SCMI_FLEXGEN_22>,
+		<&scmi_clk CK_SCMI_FLEXGEN_23>,
+		<&scmi_clk CK_SCMI_FLEXGEN_24>,
+		<&scmi_clk CK_SCMI_FLEXGEN_25>,
+		<&scmi_clk CK_SCMI_FLEXGEN_26>,
+		<&scmi_clk CK_SCMI_FLEXGEN_27>,
+		<&scmi_clk CK_SCMI_FLEXGEN_28>,
+		<&scmi_clk CK_SCMI_FLEXGEN_29>,
+		<&scmi_clk CK_SCMI_FLEXGEN_30>,
+		<&scmi_clk CK_SCMI_FLEXGEN_31>,
+		<&scmi_clk CK_SCMI_FLEXGEN_32>,
+		<&scmi_clk CK_SCMI_FLEXGEN_33>,
+		<&scmi_clk CK_SCMI_FLEXGEN_34>,
+		<&scmi_clk CK_SCMI_FLEXGEN_35>,
+		<&scmi_clk CK_SCMI_FLEXGEN_36>,
+		<&scmi_clk CK_SCMI_FLEXGEN_37>,
+		<&scmi_clk CK_SCMI_FLEXGEN_38>,
+		<&scmi_clk CK_SCMI_FLEXGEN_39>,
+		<&scmi_clk CK_SCMI_FLEXGEN_40>,
+		<&scmi_clk CK_SCMI_FLEXGEN_41>,
+		<&scmi_clk CK_SCMI_FLEXGEN_42>,
+		<&scmi_clk CK_SCMI_FLEXGEN_43>,
+		<&scmi_clk CK_SCMI_FLEXGEN_44>,
+		<&scmi_clk CK_SCMI_FLEXGEN_45>,
+		<&scmi_clk CK_SCMI_FLEXGEN_46>,
+		<&scmi_clk CK_SCMI_FLEXGEN_47>,
+		<&scmi_clk CK_SCMI_FLEXGEN_48>,
+		<&scmi_clk CK_SCMI_FLEXGEN_49>,
+		<&scmi_clk CK_SCMI_FLEXGEN_50>,
+		<&scmi_clk CK_SCMI_FLEXGEN_51>,
+		<&scmi_clk CK_SCMI_FLEXGEN_52>,
+		<&scmi_clk CK_SCMI_FLEXGEN_53>,
+		<&scmi_clk CK_SCMI_FLEXGEN_54>,
+		<&scmi_clk CK_SCMI_FLEXGEN_55>,
+		<&scmi_clk CK_SCMI_FLEXGEN_56>,
+		<&scmi_clk CK_SCMI_FLEXGEN_57>,
+		<&scmi_clk CK_SCMI_FLEXGEN_58>,
+		<&scmi_clk CK_SCMI_FLEXGEN_59>,
+		<&scmi_clk CK_SCMI_FLEXGEN_60>,
+		<&scmi_clk CK_SCMI_FLEXGEN_61>,
+		<&scmi_clk CK_SCMI_FLEXGEN_62>,
+		<&scmi_clk CK_SCMI_FLEXGEN_63>,
+		<&scmi_clk CK_SCMI_ICN_APB1>,
+		<&scmi_clk CK_SCMI_ICN_APB2>,
+		<&scmi_clk CK_SCMI_ICN_APB3>,
+		<&scmi_clk CK_SCMI_ICN_APB4>,
+		<&scmi_clk CK_SCMI_ICN_APBDBG>,
+		<&scmi_clk CK_SCMI_TIMG1>,
+		<&scmi_clk CK_SCMI_TIMG2>,
+		<&scmi_clk CK_SCMI_PLL3>,
+		<&dsi>,
+		<&scmi_clk CK_SCMI_HSI_KER_CK>,
+		<&scmi_clk CK_SCMI_HSE_KER_CK>,
+		<&scmi_clk CK_SCMI_MSI_KER_CK>;
+};
+
 &rifsc {
 	dsi: dsi@48000000 {
 		compatible = "st,stm32mp25-dsi";
